### PR TITLE
iox-#2282 Prepare v2.0.6 release

### LINKED ISSUE #2282: Prepare v2.0.6 release for ROS2
## Brief feature description

We need to create a new release with the fixes on the release_2.0 branch in order to get those fixes into the next ROS2 release

### LINKED ISSUE #2284: Compiler warnings on latest macOS on release_2.0
## Brief description

On macOS 14 there are new compiler warnings with the updated compiler

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/main/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/main/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/main/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Consider a second reviewer for complex new features or larger refactorings
- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #2282 
- Closes #2284

### CODE REVIEW

File: doc/website/release-notes/iceoryx-v2-0-6.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# iceoryx v2.0.6
```
Comment (mossmaurice) [inline doc/website/release-notes/iceoryx-v2-0-6.md:1]: @elBoberido Please have a look at this PR: https://github.com/eclipse-iceoryx/iceoryx/pull/2126/files

You need to make this file available in the MkDocs navigation by adding it to `doc/website/release-notes/.pages`

**Comment by elBoberido** [comment]
ROS2 CI
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20939/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15230/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21613/)

**Review by mossmaurice — APPROVED** [review]
LGTM, thanks for taking care of the release!

**Comment by elBoberido** [comment]
@clalancette the aarch64 builds are marked as unstable. I'm quite sure it is not related to the changes of this PR and would create the release nevertheless

**Comment by elBoberido** [comment]
@clalancette all the failed tests on aarch64 yesterday had this error message
```
mimick: Initialization error: could not find definitions for vital function(s): 'abort' 'vfprintf'
```
This is not related in any way to iceoryx and if the current run will have the same failures I'll merge this branch and create the release

**Comment by clalancette** [comment]
> This is not related in any way to iceoryx and if the current run will have the same failures I'll merge this branch and create the release

Yes, that's correct.  That's a known issue that we are tracking separately: https://github.com/ros2/Mimick/issues/35

**Pull request merged**